File: go/sprint5/J/code.go
package main

/**
Comment it before submitting 
type Node struct { 
	value    int 
	left   *Node 
	right  *Node 
}
**/

func insert(root *Node, key int) *Node {
    // Your code
    // “ヽ(´▽｀)ノ”
}

func test() {
	node1 := Node{7, nil, nil}
    node2 := Node{8, &node1, nil}
    node3 := Node{7, nil, &node2}
    newHead := insert(&node3, 6)
	if newHead != &node3 {
		panic("WA")
	}
	if newHead.left.value != 6 {
		panic("WA")
	}
}

File: python/sprint5/J/code.py
# ! change LOCAL to False before submitting !
# set LOCAL to True for local testing

LOCAL = True

if not LOCAL:
    from node import Node

if LOCAL:
    class Node:  
        def __init__(self, left=None, right=None, value=0):  
            self.right = right
            self.left = left
            self.value = value


def insert(root, key):
    #  Your code
    #  “ヽ(´▽｀)ノ”
    pass


def test():
    node1 = Node(None, None, 7)
    node2 = Node(node1, None, 8)
    node3 = Node(None, node2, 7)
    new_head = insert(node3, 6)
    assert new_head is node3
    assert new_head.left.value == 6

if __name__ == '__main__':
    test()